recipe/deactivate.sh

```diff
@@ -1,65 +1,73 @@
+# shellcheck shell=sh
 # Deactivate EUPS
 # 
 # Derived from the stackvana-core recipe by Matt Becker (GitHub @beckermr)
 # see: https://github.com/beckermr/stackvana-core/blob/master/recipe/stackvana_deactivate.sh
 #
 
-get_setup () {
+__eups_get_setup () {
     eups list -s --raw 2>/dev/null \
 	| grep -v '^eups|' | head -1 | cut -d'|' -f1 \
 	|| echo ""
 }
 
 # unsetup any products to keep env clean
-pkg=`get_setup`
-while [ -n "$pkg" ]; do
-    unsetup "$pkg" > /dev/null 2>&1
-    pkg=`get_setup`
+__eups_pkg=$(__eups_get_setup)
+while [ -n "$__eups_pkg" ]; do
+    unsetup "$__eups_pkg" > /dev/null 2>&1
+    __eups_pkg=$(__eups_get_setup)
 done
-unset pkg
-
+unset __eups_pkg
+unset -f __eups_get_setup
 
 # clean out the path, removing EUPS_DIR/bin
 # https://stackoverflow.com/questions/370047/what-is-the-most-elegant-way-to-remove-a-path-from-the-path-variable-in-bash
-# we are not using the function below because this seems to mess with conda's
-# own path manipulations
-WORK=":$PATH:"
-REMOVE=":${EUPS_DIR}/bin:"
-WORK="${WORK//$REMOVE/:}"
-WORK="${WORK%:}"
-WORK="${WORK#:}"
-export PATH="$WORK"
-unset WORK
-unset REMOVE
+# but use sed instead of bash-only regexp parameter substitution
+__EUPS_WORK=":$PATH:"
+__EUPS_REMOVE=":${EUPS_DIR}/bin:"
+# : is a safe regexp delimiter
+__EUPS_WORK=$(echo "$__EUPS_WORK" | sed -e "s:\:$__EUPS_REMOVE\::\::")
+__EUPS_WORK="${__EUPS_WORK%:}"
+__EUPS_WORK="${__EUPS_WORK#:}"
+export PATH="$__EUPS_WORK"
+unset __EUPS_WORK
+unset __EUPS_REMOVE
 
 
 # restore EUPS env variables existing prior to the activation
-for var in EUPS_PATH EUPS_SHELL SETUP_EUPS EUPS_DIR; do
-  unset $var
-  bkvar="CONDA_EUPS_BACKUP_$var"
-  eval "value=\"\${$bkvar}\""
-  if [ -n "${value}" ]; then
-    export $var="${value}"
-    unset "$bkvar"
+for __eups_var in EUPS_PATH EUPS_SHELL SETUP_EUPS EUPS_DIR; do
+  unset $__eups_var
+  __eups_bkvar="CONDA_EUPS_BACKUP_$__eups_var"
+  eval "__eups_value=\"\${$__eups_bkvar}\""
+  if [ -n "${__eups_value}" ]; then
+    export $__eups_var="${__eups_value}"
+    unset "$__eups_bkvar"
   fi
 done
-unset bkvar
-unset var
+unset __eups_bkvar
+unset __eups_var
 unset -f setup
 if [ "$CONDA_EUPS_BACKUP_setup" ]; then
   eval "$CONDA_EUPS_BACKUP_setup"
   unset CONDA_EUPS_BACKUP_setup
+  __setup_restored=1
 fi
 unset -f unsetup
-if [[ "$CONDA_EUPS_BACKUP_unsetup" ]]; then
+if [ "$CONDA_EUPS_BACKUP_unsetup" ]; then
   eval "$CONDA_EUPS_BACKUP_unsetup"
   unset CONDA_EUPS_BACKUP_unsetup
+  __setup_restored=1
```

Comment:
I think we want to explicitly check two different `__setup_restored` and `__unsetup_restored` before going into the branch below that sources eups' `setup.sh`.

Reply:
eups's `setups.sh` is going to create both `setup` and `unsetup` functions; there's no way to do one and not the other.  So if either one is restored, I assume that the user did something to them and leave them both alone.  I don't see what different path an `__unsetup_restored` would cause.

Reply:
ahhhhhh I messed up the logic in my head when I read the code. Sorry! I thought it did something else.